src/lib/search/search.ts

```diff
@@ -0,0 +1,55 @@
+import { NoteDoc } from '../db/types'
+
+export interface SearchResult {
+  id: string
+  lineStr: string
+  matchStr: string
+  matchColumn: number
+  matchLength: number
+  lineNum: number
+}
+
+export interface NoteSearchData {
+  results: SearchResult[]
+  note: NoteDoc
+}
+
+export const MAX_SEARCH_PREVIEW_LINE_LENGTH = 10000
+export const MAX_SEARCH_LINES_PER_NOTE = 10000
+export const SEARCH_DEBOUNCE_TIMEOUT = 350
+
+export function getSearchResultKey(noteId: string, searchItemId: string) {
+  return `${noteId}${searchItemId}`
+}
+
+export function getMatchData(
+  text: string,
+  searchTerm: string | RegExp
+): SearchResult[] {
+  const data: SearchResult[] = []
+
+  // Split text
+  let lines: string[] = text.split('\n')
+  let resultId = 0
+
+  // Use only first N lines
+  if (lines.length > MAX_SEARCH_LINES_PER_NOTE) {
+    lines = lines.slice(0, MAX_SEARCH_PREVIEW_LINE_LENGTH)
```

Comment:
This looks a bit weird. So we cannot search when searchTerm is including new lines?

Reply:
This was for maximum number of lines we want to search. But yes, I split the data into lines, because input element allows only one line of input, no newlines.

In new version, we provide the user with textarea which can be used to add multiline queries, works fine now if you ask me.

Also the condition is replaced with maximum amount of MBs/characters we allow to be searched per note.